### PR TITLE
Cleanup of admin console IDE warnings

### PR DESCRIPTION
**Pull request by guusdk**
This is a commit that cleans up most of the warnings that IntelliJ shows when looking at the admin console JSP pages.

Also, styling has been made more consistent were opportune: re-use of the various taglibs has been improved, and custom styling has been removed.

I've not been very careful: I've applied some global search/replaces, that could have unexpected side-effects. We should test this more than what I have done so far.

### CODE REVIEW

File: documentation/ldap-guide.html [hunk 1]
```diff
@@ -356,7 +356,7 @@ <h3>Manually Editing the Config File</h3>
 
     <pre>&lt;searchFilter&gt;&lt;![CDATA[(&(sAMAccountName={0})(|(givenName=GEORGE)(givenName=admin)))]]&gt;&lt;/searchFilter&gt;</pre>
 
-    <p><a name="ctxFactory"><h2>Custom Inital Context Factory</h2></a></p>
+    <h2 id="ctxFactory">>Custom Inital Context Factory</h2>
```
Comment (akrherz) [inline documentation/ldap-guide.html:359]: stray '>' here?

**Review by akrherz — APPROVED** [review]
Incredible Guus!  I only found one stray > so far :)

File: documentation/ldap-guide.html (hunk 1 above)
Comment (guusdk) [reply documentation/ldap-guide.html]: Good catch. Fixed it in the next PR (that addresses all files of the documentations directory).

**Comment by guusdk** [comment]
I have now applied similar changes to all HTML in the documentation folder.

File: documentation/docker.html [hunk 2]
```diff
@@ -5,29 +5,54 @@
...
+                <li><a href="#builds">Builds</a></li>
+                <li><a href="#usage">Usage</a></li>
+            </ul>
+        </nav>
+
+    </section>
+
+    <section id="builds">
+        <h2>Builds</h2>
+        <p>The official builds of Openfire for Docker come from GitHub. When a tag is pushed, a new image is built by the CI and uploaded to Docker Hub with the same tag. Any push to the main branch is built and pushed to the 'latest' tag on Docker Hub, and whilst every effort made, these should be considered potentially unstable.</p>
```
Comment (Fishbowler) [inline documentation/docker.html:39]: If it's in our docs, we should probably do this. Or change the docs.

File: xmppserver/src/main/webapp/connection-settings-advanced.jsp [hunk 3]
```diff
@@ -452,10 +452,10 @@
...
     <fmt:message key="connection.advanced.settings.misc.boxtitle" var="miscboxtitle"/>
     <admin:contentBox title="${miscboxtitle}">
-        <table cellpadding="3" cellspacing="0" border="0">
-            <tr valign="middle">
-                <td width="1%" nowrap><label for="maxThreads"><fmt:message key="connection.advanced.settings.misc.label_workers"/></label></td>
-                <td width="99%"><input type="text" name="maxThreads" id="maxThreads" value="${configuration.maxThreadPoolSize}" readonly/></td>
+        <table>
+            <tr>
+                <td style="width: 1%; white-space: nowrap"><label for="maxThreads"><fmt:message key="connection.advanced.settings.misc.label_workers"/></label></td>
+                <td><input type="text" name="maxThreads" id="maxThreads" value="${configuration.maxThreadPoolSize}" readonly/></td>
```
Comment (Fishbowler) [inline xmppserver/src/main/webapp/connection-settings-advanced.jsp:458]: Why do we have inputs that are silently readonly?

File: xmppserver/src/main/webapp/connection-settings-external-components.jsp [hunk 4]
```diff
@@ -440,30 +438,30 @@
...
                 </tr>
             </c:when>
             <c:otherwise>
                 <c:forEach var="component" varStatus="status" items="${blockedComponents}">
-                    <tr class="${ ( (status.index + 1) % 2 ) eq 0 ? 'jive-even' : 'jive-odd'}">
+                    <tr>
                         <td>${ status.index + 1}</td>
                         <td><c:out value="${component.subdomain}"/></td>
-                        <td align="center" style="border-right:1px #ccc solid;">
+                        <td style="border-right:1px #ccc solid; text-align: center">
```
Comment (Fishbowler) [inline xmppserver/src/main/webapp/connection-settings-external-components.jsp:458]: This delete icon is squiff

image

File: xmppserver/src/main/webapp/connection-settings-socket-c2s.jsp [hunk 5]
```diff
@@ -149,15 +148,15 @@
...
+        <table>
+            <tr>
+                <td colspan="2"><input type="checkbox" name="plaintext-enabled" id="plaintext-enabled" onclick="applyDisplayable('plaintext')" ${plaintextConfiguration.enabled ? 'checked' : ''}/>&nbsp;<label for="plaintext-enabled"><fmt:message key="ssl.settings.client.plaintext.label_enable"/></label></td>
             </tr>
-            <tr valign="middle">
-                <td width="1%" nowrap><label for="plaintext-tcpPort"><fmt:message key="ports.port"/></label></td>
-                <td width="99%"><input type="text" name="plaintext-tcpPort" id="plaintext-tcpPort" value="${plaintextConfiguration.port}"/></td>
+            <tr>
+                <td style="width: 1%; white-space: nowrap"><label for="plaintext-tcpPort"><fmt:message key="ports.port"/></label></td>
+                <td><input type="text" name="plaintext-tcpPort" id="plaintext-tcpPort" value="${plaintextConfiguration.port}"/></td>
```
Comment (Fishbowler) [inline xmppserver/src/main/webapp/connection-settings-socket-c2s.jsp:157]: The `valign` or similar looks needed

image

File: xmppserver/src/main/webapp/connection-settings-socket-s2s.jsp [hunk 6]
```diff
@@ -307,15 +307,15 @@
...
+        <table>
+            <tr>
+                <td colspan="2"><input type="checkbox" name="plaintext-enabled" id="plaintext-enabled" onclick="applyDisplayable('plaintext')" ${plaintextConfiguration.enabled ? 'checked' : ''}/>&nbsp;<label for="plaintext-enabled"><fmt:message key="server2server.settings.label_enable"/></label></td>
             </tr>
-            <tr valign="middle">
-                <td width="1%" nowrap><label for="plaintext-tcpPort"><fmt:message key="ports.port"/></label></td>
-                <td width="99%"><input type="text" name="plaintext-tcpPort" id="plaintext-tcpPort" value="${plaintextConfiguration.port}"/></td>
+            <tr>
+                <td style="width: 1%; white-space: nowrap"><label for="plaintext-tcpPort"><fmt:message key="ports.port"/></label></td>
+                <td><input type="text" name="plaintext-tcpPort" id="plaintext-tcpPort" value="${plaintextConfiguration.port}"/></td>
```
Comment (Fishbowler) [inline xmppserver/src/main/webapp/connection-settings-socket-s2s.jsp:316]: As above

image

File: documentation/docker.html (hunk 2 above)
Comment (guusdk) [reply documentation/docker.html:39]: I'll remove this from the docs for now.

File: xmppserver/src/main/webapp/connection-settings-external-components.jsp (hunk 4 above)
Comment (guusdk) [reply xmppserver/src/main/webapp/connection-settings-external-components.jsp:458]: The headers needs to have a similar text-align as the data.

File: xmppserver/src/main/webapp/connection-settings-socket-c2s.jsp (hunk 5 above)
Comment (guusdk) [reply xmppserver/src/main/webapp/connection-settings-socket-c2s.jsp:157]: This appears to be caused by the Bootstrap stylesheet, that adds a `margin-bottom` to labels. I'll override that for labels that are _in_ a table cell.

File: xmppserver/src/main/webapp/error-serverdown.jsp [hunk 7]
```diff
@@ -31,14 +32,14 @@
 <body>
 
 <div id="jive-header">
-<table cellpadding="0" cellspacing="0" width="100%" border="0">
+<table>
 <tbody>
     <tr>
      <td>
-         <img src="<%= path %>/<%= AdminConsole.getLogoImage() %>" border="0" alt="<%= AdminConsole.getAppName() %> <fmt:message key="error.serverdown.admin_console" />">
+         <img src="<%= path %>/<%= AdminConsole.getLogoImage() %>" alt="<%= AdminConsole.getAppName() %> <fmt:message key="error.serverdown.admin_console" />">
```
Comment (Fishbowler) [inline xmppserver/src/main/webapp/error-serverdown.jsp:39]: `getLogoImage()` appears to default to `/images/header-title.gif`, but this image appears missing in my docker container.
I can't find it in the repo, but possibly I'm being daft, or it's renamed in the build, or it's just an access path for a different file on disk :man_shrugging:

File: xmppserver/src/main/webapp/group-edit.jsp [hunk 8]
```diff
@@ -570,12 +569,12 @@
...
                 <tr>
-                    <td nowrap width="1%">
+                    <td style="width: 1%; white-space: nowrap">
                         <fmt:message key="group.edit.add_user" />
                     </td>
-                    <td nowrap class="c1" align="left">
+                    <td class="c1" style="text-align: left; white-space: nowrap">
                          <input type="text" size="45" name="username"/>
                         &nbsp;<input type="submit" name="addbutton" value="<fmt:message key="global.add" />">
                     </td>
```
Comment (Fishbowler) [inline xmppserver/src/main/webapp/group-edit.jsp:580]: The add member text input is selected by default when entering the group edit page.
Is this intentional behaviour?
Oh, it's pre-existing.

File: xmppserver/src/main/webapp/error-serverdown.jsp (hunk 7 above)
Comment (guusdk) [reply xmppserver/src/main/webapp/error-serverdown.jsp:40]: wow... last touched in 2006 - and I never even saw this. I don't think anyone sees that JSP, ever?

Apparently, this is taken from `admin-sidebar.xml`, which has references to old wildfire stuff. This was already failing prior to this PR, by the way.

I have removed the references from `admin-sidebar.xml`, and added an `if` statement to this usage, which will prevent the image from being (attempted to be) displayed, when the reference is not there.

Modifications made in #ff7e2d1211264d7c442a401ba0cf2cec81b9d236

File: xmppserver/src/main/webapp/group-edit.jsp (hunk 8 above)
Comment (guusdk) [reply xmppserver/src/main/webapp/group-edit.jsp:580]: I think it _is_ intentional, as that's the page that is shown directly after a group is created. It stands to reason that after you create a group, you want to add members to it.

**Comment by Fishbowler** [comment]
Seeing a javascript error on the admin index page. I don't think this is new, but don't know if this is something that's best/asily tackled here:

```
Uncaught TypeError: Behaviour is not iterable
    at Object.apply (behaviour.js:54:29)
    at behaviour.js:49:23
```

**Comment by Fishbowler** [comment]
/loginToken.jsp looking misformatted:

image

Main branch looks different (but equally not superb)

image

**Comment by Fishbowler** [comment]
Another pre-existing bug that I'll list here, and split to Jira if you can't / don't fancy bundling it into this:

`/muc-create-permission.jsp` for a service (e.g. /muc-create-permission.jsp?mucname=conference on demoboot) gives a JS error when you pick "Only specific..."

```
Uncaught TypeError: Cannot read properties of undefined (reading 'focus')
    at HTMLInputElement.onfocus (muc-create-permission.jsp?mucname=conference:203:41)
onfocus @ muc-create-permission.jsp?mucname=conference:203
```

**Comment by guusdk** [comment]
> /loginToken.jsp looking misformatted:

I have now pushed a change that should bring it in-line with the regular 'login' screen.

**Comment by guusdk** [comment]
> Seeing a javascript error on the admin index page. I don't think this is new, but don't know if this is something that's best/asily tackled here

I think that this is pre-existing. I'd prefer to tackle it as a separate issue.

**Comment by guusdk** [comment]
> Another pre-existing bug that I'll list here, and split to Jira if you can't / don't fancy bundling it into this:

Well, I can get rid of the error by removing the line that requests focus on the not-displayed input field, but I think that the proper fix here is to dynamically show that user selection field (and give focus to it).

Can you raise an issue for this one please?

**Comment by Fishbowler** [comment]
One more pre-existing issue, this time in `/offline-messages.jsp` that's also broken in `main` - we can't save the offline settings if you set the Per-user offline message storage limit to a negative number. It's looks minor, but _might_ be unrecoverable

```
Exception:
javax.el.MethodNotFoundException: Method not found: class java.lang.String.toMinutes()
	at javax.el.Util.findWrapper(Util.java:373)
	at javax.el.Util.findMethod(Util.java:219)
	at javax.el.BeanELResolver.invoke(BeanELResolver.java:149)
	at org.apache.jasper.el.JasperELResolver.invoke(JasperELResolver.java:151)
	at org.apache.el.parser.AstValue.getValue(AstValue.java:159)
	at org.apache.el.ValueExpressionImpl.getValue(ValueExpressionImpl.java:190)
	at org.apache.jasper.runtime.PageContextImpl.proprietaryEvaluate(PageContextImpl.java:944)
	at org.jivesoftware.openfire.admin.offline_002dmessages_jsp._jspService(offline_002dmessages_jsp.java:419)
	at org.apache.jasper.runtime.HttpJspBase.service(HttpJspBase.java:71)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
	at org.eclipse.jetty.servlet.ServletHolder$NotAsync.service(ServletHolder.java:1459)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:799)
	at org.eclipse.jetty.servlet.ServletHandler$ChainEnd.doFilter(ServletHandler.java:1626)
	at com.opensymphony.sitemesh.webapp.SiteMeshFilter.obtainContent(SiteMeshFilter.java:129)
	at com.opensymphony.sitemesh.webapp.SiteMeshFilter.doFilter(SiteMeshFilter.java:77)
	at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193)
	at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601)
	at org.jivesoftware.util.LocaleFilter.doFilter(LocaleFilter.java:73)
	at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193)
	at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601)
	at org.jivesoftware.util.SetCharacterEncodingFilter.doFilter(SetCharacterEncodingFilter.java:49)
	at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193)
	at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601)
	at org.jivesoftware.admin.PluginFilter.doFilter(PluginFilter.java:174)
```

**Comment by Fishbowler** [comment]
Right. Happy with all of the Admin Console changes up to this point. Either I've mentioned them above, tested them and they're fine, or not tested them because the diff looked harmless.

i18n and docs to go.

**Comment by guusdk** [comment]
> One more pre-existing issue, this time in `/offline-messages.jsp` that's also broken in `main`

Raised as https://igniterealtime.atlassian.net/browse/OF-2552 and pushed a fix to this PR. This was a fun one: because of the error that was introduced, a _different_ error was triggered. No invalid value was ever persisted, as far as I can tell, so I do not believe that this was unrecoverable.

File: documentation/database.html [hunk 9]
```diff
@@ -1,324 +1,451 @@
...
+                            <code>createdb -E UNICODE -O openfireuser openfire</code>
+                        </li>
+                    </ul>
+                </li>
+                <li>
+                    Proceed with Openfire install setup via the admin console, the setup process with install the schema.
+                </li>
+            </ol>
+
+        </section>
```
Comment (Fishbowler) [inline documentation/database.html:323]: Worth adding a useful link where we can, like with mysql?

```suggestion
                <li>
                    Proceed with Openfire install setup via the admin console, the setup process with install the schema.
                </li>
            </ol>

        </section>
         <section id="postgres-further-help">

            <h3>Further Help</h3>

            <p>
                If you need help setting up Postgres, refer to the following site:
            </p>
            <ul>
                <li><a href=["https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-reference-configuration-properties.html](https://jdbc.postgresql.org/documentation/use/#connecting-to-the-database)">Postgres Configuration and Connection documentation</a></li>
            </ul>

        </section>
```

File: documentation/db-clustering-guide.html [hunk 10]
```diff
@@ -0,0 +1,275 @@
...
+        <p>
+            The <a href="database.html">Database Installation Guide</a> describes how Openfire is configured to connect
+            to a database. Primarily, a <abbr title="Java DataBase Connectivity">JDBC</abbr> URL is used to define
+            most of the configuration. This URL varies depending on your database vendor. These are some examples:
+        </p>
+        <dl>
+            <dt>MySQL</dt><dd><code>jdbc:mysql://HOSTNAME:3306/DATABASENAME</code></dd>
+            <dt>Oracle</dt><dd><code>jdbc:oracle:thin:@HOSTNAME:1521:SID</code></dd>
+            <dt>Microsoft SQL Server</dt><dd><code>jdbc:sqlserver://HOSTNAME:1433;databaseName=DATABASENAME;applicationName=Openfire</code></dd>
+        </dl>
```
Comment (Fishbowler) [inline documentation/db-clustering-guide.html:126]: Add PostgreSQL

```suggestion
        <dl>
            <dt>MySQL</dt><dd><code>jdbc:mysql://HOSTNAME:3306/DATABASENAME</code></dd>
            <dt>Oracle</dt><dd><code>jdbc:oracle:thin:@HOSTNAME:1521:SID</code></dd>
            <dt>Microsoft SQL Server</dt><dd><code>jdbc:sqlserver://HOSTNAME:1433;databaseName=DATABASENAME;applicationName=Openfire</code></dd>
            <dt>PostgreSQL</dt><dd><code>jdbc:postgresql://HOSTNAME:5432/DATABASENAME</code></dd>
        </dl>
```

File: documentation/db-clustering-guide.html [hunk 11]
```diff
@@ -0,0 +1,275 @@
...
+
+            <h4>Links to vendor-specific documentation</h4>
+
+            <ul>
+                <li><a href="https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-multi-host-connections.html">MySQL Connector/J 8.0 Developer Guide: Multi-Host Connections</a></li>
+                <li><a href="https://docs.oracle.com/en/database/oracle/oracle-database/21/jjdbc/application-continuity.html#GUID-9298BB8C-0E93-44BF-BA26-A615940C0C84">Oracle Database
+                    JDBC Developer's Guide and Reference: About Configuring Oracle JDBC for Application Continuity for Java</a></li>
+                <li><span style="color: red;">TBD: complete this for all (?) supported vendors.</span></li>
+            </ul>
+
```
Comment (Fishbowler) [inline documentation/db-clustering-guide.html:155]: Add PostgreSQL docs link

```suggestion
            <ul>
                <li><a href="https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-multi-host-connections.html">MySQL Connector/J 8.0 Developer Guide: Multi-Host Connections</a></li>
                <li><a href="https://docs.oracle.com/en/database/oracle/oracle-database/21/jjdbc/application-continuity.html#GUID-9298BB8C-0E93-44BF-BA26-A615940C0C84">Oracle Database
                    JDBC Developer's Guide and Reference: About Configuring Oracle JDBC for Application Continuity for Java</a></li>
                <li><a href="https://jdbc.postgresql.org/documentation/use/#connection-fail-over">PostgreSQL JDBC: Connection failover</a></li>  
                <li><span style="color: red;">TBD: complete this for all (?) supported vendors.</span></li>
            </ul>

```

File: documentation/ide-vscode-setup.html [hunk 12]
```diff
@@ -3,187 +3,204 @@
...
+        <section id="build-openfire-from-vs-code">
+
+            <h2>Build Openfire from VS Code</h2>
+            <p>
+                The following instructions will explain how you can build Openfire from the code that is in the VS Code workspace.
+            </p>
+            <p>
+                Select the 'New Terminal' option -- <b>Terminal > New Terminal</b> to open a terminal and run the command bellow to  build the complete project including plugins:
+            </p>
+            <pre>mvn clean verify</pre>
```
Comment (Fishbowler) [inline documentation/ide-vscode-setup.html:159]: s/mvn/mvnw

```suggestion
            <pre>./mvnw clean verify</pre>
```

File: documentation/install-guide.html [hunk 13]
```diff
@@ -1,145 +1,250 @@
...
+
+            <p>
+                As an alternative to the web-based "wizard" driven setup described above, Openfire can be configured by
+                provisioning it with a specially crafted <code>openfire.xml</code> file.
+            </p>
+
+            <p>
+                If <code>&lt;setup&gt;true&lt;>/setup&gt;</code> does not exist in <code>openfire.xml</code> (i.e. setup is not
+                complete, and a block does exist called <code>&lt;autosetup&gt;</code> with a <code>&lt;run&gt;true&lt;/run&gt;</code>
+                property inside it, that looks like this:
```
Comment (Fishbowler) [inline documentation/install-guide.html:242]:
```suggestion
                If <code>&lt;setup&gt;true&lt;>/setup&gt;</code> does not exist in <code>openfire.xml</code> (i.e. setup is not
                complete), and a block <em>does</em> exist called <code>&lt;autosetup&gt;</code> with a <code>&lt;run&gt;true&lt;/run&gt;</code>
                property inside it, that looks like this:
```

File: documentation/install-guide.html [hunk 14]
```diff
@@ -191,158 +296,243 @@ <h3><a name="autosetup"></a>Autosetup</h3>
...
+        </p>
+
+        <section>
+
+            <h3>Windows Service</h3>
+
+            <figure class="inline-right">
+                <img src="images/windows_service.png" alt="Windows Service" width="557" height="223">
+                <figcaption>Windows Service commands in a console windows.</figcaption>
+            </figure>
```
Comment (Fishbowler) [inline documentation/install-guide.html:349]: I'll try and source a replacement image for this

File: documentation/install-guide.html [hunk 15]
```diff
@@ -191,158 +296,243 @@ <h3><a name="autosetup"></a>Autosetup</h3>
...
+                to 512 MB and max VM heap size to 1024 MB, you'd use:
+            </p>
+
+            <pre><code>-Xms512m
+-Xmx1024m</code></pre>
+
+            <p>
+                To create parameters for the normal launcher, create a file called <code>openfired.vmoptions</code>
+                (since the openfire.exe launcher invokes the openfired.exe executable to actually start the server).
+            </p>
```
Comment (Fishbowler) [inline documentation/install-guide.html:409]: Probably out of scope, but it's not clear which file I would normally want to add to increase the memory cap for a running instance, and I'd probably punt and set all three.

File: documentation/install-guide.html [hunk 16]
```diff
@@ -191,158 +296,243 @@ <h3><a name="autosetup"></a>Autosetup</h3>
...
+        <h2>Firewall</h2>
+
+        <p>
+            Server's administrator should open TCP ports 5222 and 7443 for incoming connections for XMPP clients to be
+            able to connect to Openfire. Additionally, an administrator can also open TCP 9090 (for http) and TCP 9091
+            (for https), if there is a need to remotely administrate Openfire connecting to its Admin Console. We
+            recommend only using 9091 port as it is an encrypted connection. For server to server connections one should
+            also open TCP 5269 port and for secure HTTP-BIND connections TCP 7443 port. Port number can be different, if
+            the default configuration has been changed by an administrator. All ports used by Openfire can be found on
+            the first page of Admin Console in the Server Ports section.
```
Comment (Fishbowler) [inline documentation/install-guide.html:523]:
```suggestion
            the default configuration has been changed by an administrator. Additional ports may also be in use by 
            Openfire or by plugins to provide additional features. The full list of ports used by Openfire can be found on
            the first page of Admin Console in the Server Ports section.
```

File: documentation/ldap-guide.html [hunk 17]
```diff
@@ -1,287 +1,357 @@
...
+        <h2>Background</h2>
+
+        <p>
+            LDAP (Lightweight Directory Access Protocol) has emerged as a dominant standard
+            for user authentication and for storage of user profile data. It serves as a
+            powerful tool for large organizations (or those organizations integrating many
+            applications) to simplify user management issues. Many LDAP servers are available,
+            such as <a href="https://www.openldap.org/">Open LDAP</a>,
+            <a href="https://www.microsoft.com/windowsserver2003/technologies/directory/activedirectory/">Active Directory</a>,
+            and Novell's <a href="https://www.novell.com/products/edirectory/">eDirectory</a>.
```
Comment (Fishbowler) [inline documentation/ldap-guide.html:60]: Remove this reference to Novell? Looks like it's been sold on, there's no trial that we can test with, and we've no idea if it'll work nowadays.

File: documentation/ldap-guide.html [hunk 18]
```diff
@@ -1,287 +1,357 @@
...
+    <section id="background">
+
+        <h2>Background</h2>
+
+        <p>
+            LDAP (Lightweight Directory Access Protocol) has emerged as a dominant standard
+            for user authentication and for storage of user profile data. It serves as a
+            powerful tool for large organizations (or those organizations integrating many
+            applications) to simplify user management issues. Many LDAP servers are available,
+            such as <a href="https://www.openldap.org/">Open LDAP</a>,
```
Comment (Fishbowler) [inline documentation/ldap-guide.html:58]: Typo

```suggestion
            such as <a href="https://www.openldap.org/">OpenLDAP</a>,
```

File: documentation/ldap-guide.html [hunk 19]
```diff
@@ -310,212 +380,259 @@ <h3>Manually Editing the Config File</h3>
...
+            </li>
+            <li>
+                You may have set the baseDN to an incorrect value. The LDAP module recursively searches for users under
+                the node in the directory specified by the baseDN. When the baseDN is incorrect, no users will be found.
+            </li>
+        </ol>
+        <p>
+            You can also enable debugging to get more information from the LDAP module. To do this, add
+            <code>&lt;log&gt;&lt;debug&gt;&lt;enabled&gt;true&lt;/enabled&gt;&lt;/debug&gt;&lt;/log&gt;</code> to your
+            <code>conf/openfire.xml</code> file. Log statements will be written to the <code>logs/debug.log</code> file.
```
Comment (Fishbowler) [inline documentation/ldap-guide.html:623]: Fix log filename

```suggestion
            <code>conf/openfire.xml</code> file. Log statements will be written to the <code>logs/openfire.log</code> file.
```

File: documentation/overlay.html [hunk 20]
```diff
@@ -86,19 +103,33 @@ <h2><a name="mysql">File Structure</a></h2>
...
 
+    </section>
+
+    <section id="build">
+
         <h2>Build</h2>
-        <p>Simply build the application as normal.  You may need to run the ant <tt>clean</tt>
+
+        <p>
+            Simply build the application as normal. You may need to run the ant <code>clean</code>
```
Comment (Fishbowler) [inline documentation/overlay.html:119]: Fix old ant reference. Are ant and maven comparable for 'clean'?

```suggestion
            Simply build the application as normal. You may need to run the <code>./mvnw clean</code>
```

File: documentation/plugin-dev-guide.html [hunk 21]
```diff
@@ -266,137 +314,215 @@ <h2>Modifying the Admin Console</h2>
...
-<pre>
-   &lt;html&gt;
+    </section>
+
+    <section id="writing-pages">
+
+        <h2>Writing Pages for the Admin Console</h2>
+
+        <p>
+            Openfire uses the <a href="https://www.opensymphony.com/sitemesh/" target="_blank">Sitemesh</a>
```
Comment (Fishbowler) [inline documentation/plugin-dev-guide.html:370]: Is the OpenSymphony website gone?

File: documentation/source-build.html [hunk 22]
```diff
@@ -1,253 +1,240 @@
...
+                invoke the build tool to compile the Openfire source code:
+            </p>
+            <pre>mvn compile</pre>
+            <p>
+                If the build tool is invoked correctly and Openfire compiles, you've correctly
+                configured Maven and your copy of the Openfire source distribution.
+            </p>
 
-<span class="code"> mvn clean <br></span>
+        </section>
```
Comment (Fishbowler) [inline documentation/source-build.html:142]: Should we ditch the maven section now that we've got the wrappers properly in place?

Comment (Fishbowler) [reply documentation/source-build.html:142]: (which reminds be that buidWithDocker.sh is almost certainly not using the wrapper - must create a ticket for this)

File: documentation/source-build.html [hunk 23]
```diff
@@ -1,253 +1,240 @@
...
+            <h3>compile</h3>
 
-<ul>
-<i>Syntax:</i><p>
+            <p>Builds Openfire components into the <code>target</code> directory of each module.</p>
 
-<span class="code"> mvn test <br></span>
+            <fieldset>
+                <legend>Syntax</legend>
+                <pre><code>mvn compile</code></pre>
```
Comment (Fishbowler) [inline documentation/source-build.html:192]:
```suggestion
                <pre><code>./mvnw compile</code></pre>
```

File: documentation/source-build.html [hunk 24]
```diff
@@ -1,253 +1,240 @@
...
+            <h3>test</h3>
 
-</ul>
+            <p>Builds Openfire components, and runs all unit tests.</p>
 
-<!--TEST-->
-<p><b><a name="package">package</a></b>
+            <fieldset>
+                <legend>Syntax</legend>
+                <pre><code>mvn test</code></pre>
```
Comment (Fishbowler) [inline documentation/source-build.html:205]:
```suggestion
                <pre><code>./mvnw test</code></pre>
```

File: documentation/source-build.html [hunk 25]
```diff
@@ -1,253 +1,240 @@
...
+            <p>
+                Builds Openfire components, runs all unit tests, and creates a distributable package
+                in the <code>distribution</code> module. You can then launch Openfire using the scripts in
+                <code>distribution/target/distribution-base/bin</code>.
+            </p>
 
-</ul>
+            <fieldset>
+                <legend>Syntax</legend>
+                <pre><code>mvn package</code></pre>
```
Comment (Fishbowler) [inline documentation/source-build.html:223]:
```suggestion
                <pre><code>./mvnw package</code></pre>
```

File: documentation/working-with-openfire.html [hunk 26]
```diff
@@ -1,47 +1,97 @@
...
+    </section>
+
+    <section>
+
+        <h2>Tooling</h2>
+
+        <section id="tooling-docker">
+
+            <h3>Docker</h3>
+
```
Comment (Fishbowler) [inline documentation/working-with-openfire.html:60]: Should the docker tips be merged with / linked to the main docker page?
It's out of scope for this - another ticket to be made?

**Comment by Fishbowler** [comment]
Done!
We need to tidy up the clustering WIP, and we're ready to merge

File: documentation/db-clustering-guide.html [hunk 27]
```diff
@@ -0,0 +1,275 @@
...
+            vendor. The addition of cluster-specific configuration such as the definition of multiple hosts, or the
+            desired replication type, adds even more variation. When you want to define the JDBC URL that is used to
+            configure Openfire for database clustering, you will need to refer to the vendor-provided documentation.
+            Some example JDBC URLs that define more than one database server, as well as additional configuration:
+        </p>
+        <dl>
+            <dt>MySQL</dt><dd><code>jdbc:mysql://PRIMARYHOSTNAME:3306,SECONDARYHOST:3306/DATABASENAME?failOverReadOnly=false</code></dd>
+            <dt>Oracle</dt><dd><span style="color: red;">TBD</span></dd>
+            <dt>Microsoft SQL Server</dt><dd><span style="color: red;">TBD</span></dd>
+        </dl>
```
[the remark this answers is not in the document]
Comment (guusdk) [reply documentation/db-clustering-guide.html:144]: I'm not sure about that `targetServerType` argument. I expect that this is very specific to one of the servers in the cluster?

File: documentation/ldap-guide.html (hunk 17 above)
Comment (guusdk) [reply documentation/ldap-guide.html:60]: Agreed.

File: documentation/overlay.html (hunk 20 above)
Comment (guusdk) [reply documentation/overlay.html:119]: Accidentally, yes.

File: documentation/plugin-dev-guide.html (hunk 21 above)
Comment (guusdk) [reply documentation/plugin-dev-guide.html:370]: Yeah, I noticed that too. I don't immediately have a credible replacement.

File: documentation/source-build.html (hunk 22 above)
Comment (guusdk) [reply documentation/source-build.html:142]: I wouldn't be surprised if most use the non-wrapper anyway (I know I do). Maybe we should _add_ the wrapper to this text?

File: documentation/install-guide.html (hunk 15 above)
Comment (guusdk) [reply documentation/install-guide.html:409]: Doesn't hurt to do that :) I'm not sure how to improve this without adding so much text that it becomes unreadable. Feel free to add a change, but I'm happy to keep this as-is.

File: documentation/working-with-openfire.html (hunk 26 above)
Comment (guusdk) [reply documentation/working-with-openfire.html:60]: I have no strong opinion either way. If we do go for that, lets indeed take it on under a different ticket.

File: documentation/db-clustering-guide.html (hunk 27 above)
[the remark this answers is not in the document]
Comment (Fishbowler) [reply documentation/db-clustering-guide.html:144]: In a PostgreSQL cluster, they elect their own primary, and you want to be targetting write ops to that one.

File: documentation/working-with-openfire.html (hunk 26 above)
Comment (Fishbowler) [reply documentation/working-with-openfire.html:60]: https://igniterealtime.atlassian.net/browse/OF-2554

File: documentation/install-guide.html (hunk 14 above)
Comment (Fishbowler) [reply documentation/install-guide.html:349]: Done, but I thought the background text was part of an old screenshot, but it isn't. It's a formatting problem on install-guide where the descriptions are being squished between the terms and the screenshot.

image


I'd offer to fix it, but I'd have no clue where to begin. Sorry...

File: documentation/ldap-guide.html (hunk 17 above)
Comment (Fishbowler) [reply documentation/ldap-guide.html]: Fixed

File: documentation/install-guide.html (hunk 14 above)
Comment (guusdk) [reply documentation/install-guide.html]: Ah, thanks for updating that! I think that I've fixed the alignment-issue. Force-pushed the fix to your last commit.

**Review by Fishbowler — APPROVED** [review]

**Comment by guusdk** [comment]
Unsure why Smack tests are flappy. Given the nature of the changes in this PR, I don't think it's related. Will merge regardless, and follow up with a new PR to re-introduce the new Database Clustering Guide (which is a work in progress).

**Pull request merged**